MoreLinq/Create.cs

```diff
@@ -0,0 +1,48 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2017 Atif Aziz. All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+
+    static partial class MoreEnumerable
+    {
+        /// <summary>
+        /// Creates an <see cref="IEnumerable{T}"/> given a factory function
+        /// for an <see cref="IEnumerator{T}"/>.
+        /// </summary>
+        /// <typeparam name="T">Type of returned sequence elements.</typeparam>
+        /// <param name="factory">
+        /// Function that creates an <see cref="IEnumerator{T}"/>.</param>
+        /// <remarks>
+        /// This method uses deferred execution semantics and streams its
+        /// results.
+        /// </remarks>
+
+        public static IEnumerable<T> Create<T>(Func<IEnumerator<T>> factory)
```

Comment:
> OOC, what other operators?

High-level operators that summarize when you are exploring a data set, somewhat akin to [`summary` in R](https://stat.ethz.ch/R-manual/R-devel/library/base/html/summary.html):

```
> summary(cars)
     speed           dist       
 Min.   : 4.0   Min.   :  2.00  
 1st Qu.:12.0   1st Qu.: 26.00  
 Median :15.0   Median : 36.00  
 Mean   :15.4   Mean   : 42.98  
 3rd Qu.:19.0   3rd Qu.: 56.00  
 Max.   :25.0   Max.   :120.00 
```

Also posted an example to #291 but they're really half-baked ideas so not ready to roll into MoreLINQ and possibly even out of scope (like we decided for linear regression from #298).

> If we have something that builds on top of `Create`, it would be weird if we didn't provide it.

True though most won't care how one operator uses another internally. It doesn't discount `Create` to stand on its own. I was eager to add this for further exploration and while I'm perfectly happy with using the one from System.Interactive, it causes conflicts with MoreLinq in some cases (pending #214).

> I wonder if we should just bite the bullet and do the same for MoreLinq. It would save [having to create single-purpose, internal structs](https://github.com/morelinq/MoreLINQ/pull/287/files#diff-d45d5bd92cdeef152abc5d8a68122510R89), and would eliminate a couple of `#if`s.

Let's take that up in a new issue.